scene/animation/animation_player.cpp

```diff
@@ -1196,9 +1195,14 @@ void AnimationPlayer::play(const StringName &p_name, float p_custom_blend, float
 
 	if (c.assigned != name) { // reset
 		c.current.pos = p_from_end ? c.current.from->animation->get_length() : 0;
-	} else if (p_from_end && c.current.pos == 0) {
-		// Animation reset BUT played backwards, set position to the end
-		c.current.pos = c.current.from->animation->get_length();
+	} else {
+		if (p_from_end && c.current.pos == 0) {
+			// Animation reset BUT played backwards, set position to the end
+			c.current.pos = c.current.from->animation->get_length();
+		} else if (c.current.pos == c.current.from->animation->get_length()) {
```

Comment:
Should it also check `!p_from_end`? Otherwise I guess it could still fail if you play the animation once normally (so position is at the end), and then you try to play it backwards.

Reply:
Yes, don't know how I didn't see it. Now checking.